### DISCUSSION TITLE
error when trying to shorten link, using CLI

### DISCUSSION THREAD

**Discussion by mrlp57** [shlinkio/discussions#108] [Help wanted] [answered]
setup selfhosted shlink on Ubuntu with lamp stack.

when using cli to shorten link, I get the error listed below.
I hope there is an answer for this, I really want to use cli

PHP Fatal error:  Uncaught Error: Undefined constant "IDNA_DEFAULT" in /var/www/shlink/vendor/guzzlehttp/guzzle/src/Client.php:217
Stack trace:
#0 /var/www/shlink/vendor/guzzlehttp/guzzle/src/Client.php(160): GuzzleHttp\Client->buildUri()
#1 /var/www/shlink/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Client->requestAsync()
#2 /var/www/shlink/module/Core/src/Util/UrlValidator.php(69): GuzzleHttp\Client->request()
#3 /var/www/shlink/module/Core/src/Util/UrlValidator.php(56): Shlinkio\Shlink\Core\Util\UrlValidator->validateUrlAndGetResponse()
#4 /var/www/shlink/module/Core/src/ShortUrl/Helper/ShortUrlTitleResolutionHelper.php(25): Shlinkio\Shlink\Core\Util\UrlValidator->validateUrlWithTitle()
#5 /var/www/shlink/module/Core/src/Service/UrlShortener.php(49): Shlinkio\Shlink\Core\ShortUrl\Helper\ShortUrlTitleResolutionHelper->processTitleAndValidateUrl()
#6 /var/www/shlink/module/CLI/src/Command/ShortUrl/GenerateShortUrlCommand.php(152): Shlinkio\Shlink\Core\Service\UrlShortener->shorten()
#7 /var/www/shlink/vendor/symfony/console/Command/Command.php(256): Shlinkio\Shlink\CLI\Command\ShortUrl\GenerateShortUrlCommand->execute()
#8 /var/www/shlink/vendor/symfony/console/Application.php(971): Symfony\Component\Console\Command\Command->run()
#9 /var/www/shlink/vendor/symfony/console/Application.php(290): Symfony\Component\Console\Application->doRunCommand()
#10 /var/www/shlink/vendor/symfony/console/Application.php(166): Symfony\Component\Console\Application->doRun()
#11 /var/www/shlink/config/run.php(14): Symfony\Component\Console\Application->run()
#12 /var/www/shlink/bin/cli(7): {closure}()
#13 {main}
  thrown in /var/www/shlink/vendor/guzzlehttp/guzzle/src/Client.php on line 217

**Comment by acelaya** [comment, marked as answer]
Make sure you follow the instructions here when installing Shlink https://shlink.io/documentation/install-dist-file/

In this case, this error means you are missing the intl extension.

**Reply by linuxmel57** [reply]
Thanks for the fast reply. I'll research to install extension.
